lib/std/heap.zig

```diff
@@ -721,12 +802,46 @@ test "c_allocator" {
     }
 }
 
+test "WasmPageAllocator internals" {
```

Comment:
I just realized this test can easily fail if any other file depends on page_allocator. Unfortunately, it looks pretty deep into the internal (unresetable) global state to make sure things are sane.

I have a few ideas to make this _less_ dependent on existing behavior, but it'll have to assume total control of all global memory. Maybe that's okay?

Reply:
I think PageAllocator assuming total control of all wasm-intrinsically allocated memory is a reasonable design choice.

Reply:
I rewrote these tests to have much fewer assumptions on the internals. There's a hard dependency on ~270 MB of execution memory but I don't think it's *too* bad since it'll be reused.